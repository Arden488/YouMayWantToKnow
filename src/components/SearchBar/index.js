import React from 'react';
import TextField from '@material-ui/core/TextField';

const SearchBar = () => (
  <div>
    <TextField
      label="Search for something"
      placeholder="Something like... World Cup 2018"
      fullWidth
      margin="normal"
    />
  </div>
);

export default SearchBar;