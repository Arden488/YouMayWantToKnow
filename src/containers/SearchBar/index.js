import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
// import { compose } from 'recompose';

import { fetchData } from '../../actions';

import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import SearchIcon from '@material-ui/icons/Search';
import Grid from '@material-ui/core/Grid';

const styles = theme => ({
  root: {
    margin: '50px auto 50px',
    [theme.breakpoints.up('lg')]: {
      width: 1170,
    },
  },
  input: {
    flexGrow: 1,
  }
});

class SearchBar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ''
    }

    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      value: nextProps.query,
    })
  } 

  render() {
    const { classes } = this.props;

    return (
      <div>
        <form onSubmit={this.handleSearchSubmit}>
          <Grid container spacing={8} className={classes.root} alignItems="baseline">
            <Grid item className={classes.input}>
              <TextField
                label="Search for something"
                placeholder="Something like... World Cup 2018"
                fullWidth
                margin="normal"
                value={this.state.value}
                onChange={this.handleInputChange}
              />
            </Grid>
            <Grid item>
              <Button 
                variant="fab" 
                mini
                color="primary" 
                aria-label="search"
                type="submit"
              >
                <SearchIcon />
              </Button>
            </Grid>
          </Grid>
        </form>
      </div>
    );
  }

  handleInputChange(event) {
    this.setState({
      value: event.target.value
    });
  } 

  handleSearchSubmit(event) {
    event.preventDefault();

    if(this.state.value != '')
      this.props.fetchData(this.state.value);
  }
}

SearchBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    query: state.query,
  };
}

function mapDispatchToProps(dispatch) {
  return { 
    fetchData: (query) => dispatch(fetchData(query)),
  };
}

export default compose(
  withStyles(styles),
  connect(mapStateToProps, mapDispatchToProps),
)(SearchBar);