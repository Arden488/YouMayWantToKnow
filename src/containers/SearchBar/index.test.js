import React from 'react';
import SearchBar from './index';
import { createShallow } from '@material-ui/core/test-utils';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

describe('<SearchBar />', () => {
  let store;
  let component;

  beforeAll(() => {
    const middlewares = [ thunk ];
    const mockStore = configureStore(middlewares);
    store = mockStore({});
    shallow = createShallow({ dive: true });
    component = shallow(<SearchBar store={store} />).dive();
  })

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('changes state on value change', () => {
    const value = 'Test';

    component.find('TextField').simulate('change', {
      target: {
        value
      }
    });

    expect(component.state().value).toEqual(value);
  });

  it('submits form', () => {
    const value = 'Test';
    
    component.find('TextField').simulate('change', {
      target: {
        value
      }
    });

    component.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
  });
});