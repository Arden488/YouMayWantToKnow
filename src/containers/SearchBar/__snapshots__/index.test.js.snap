// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(Grid)
      alignItems="baseline"
      className="Connect-SearchBar--root-1"
      container={true}
      spacing={8}
    >
      <WithStyles(Grid)
        className="Connect-SearchBar--input-2"
        item={true}
      >
        <TextField
          fullWidth={true}
          label="Search for something"
          margin="normal"
          onChange={[Function]}
          placeholder="Something like... World Cup 2018"
          required={false}
          select={false}
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
      >
        <WithStyles(Button)
          aria-label="search"
          color="primary"
          mini={true}
          type="submit"
          variant="fab"
        >
          <pure(Search) />
        </WithStyles(Button)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </form>
</div>
`;
